specifyweb/frontend/js_src/lib/components/TreeView/CreateTree.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ *WIP File to separate out functions for adding trees.
+ *Contains stuff for dialogs, etc.
+ *
+ *TODO:
+ *- Add localizations for the tree default names
+ *- Add page reload upon save
+ *- Remove comments
+ */
+import React from 'react';
+
+import { commonText } from '../../localization/common';
+import { treeText } from '../../localization/tree';
+import { DeepPartial } from '../../utils/types';
+import { Ul } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import { className } from '../Atoms/className';
+import type { AnySchema, AnyTree , SerializedResource} from '../DataModel/helperTypes';
+import { SpecifyResource } from '../DataModel/legacyTypes';
+import { deserializeResource } from '../DataModel/serializers';
+import { TaxonTreeDef } from '../DataModel/types';
+import { ResourceView } from '../Forms/ResourceView';
+import { userInformation } from '../InitialContext/userInformation';
+import { Dialog } from '../Molecules/Dialog';
+import { defaultTreeDefs } from './TreeDefaults';
+
+export function CreateTree<SCHEMA extends AnyTree>({
+  tableName,
+}: {
+  readonly tableName: SCHEMA['tableName'];
+}): JSX.Element {
+
+  const [isActive, setIsActive] = React.useState(0);
+  const [selectedResource, setSelectedResource] = React.useState<SpecifyResource<AnySchema> | null>(null);
+
+  const handleClick = (resource: DeepPartial<SerializedResource<TaxonTreeDef>>) => {
+    const dsResource = deserializeResource(resource);
+    setSelectedResource(dsResource);
+    setIsActive(2);
+  };
+
+  return (
+    <>
+      {tableName === 'Taxon' && userInformation.isadmin ? (
+        <Button.Icon
+          className={className.dataEntryAdd}
+          icon="plus"
+          title={treeText.addTree()}
+          onClick={() => {
+            setIsActive(1);
+          }}
+        />
+      ) : null}
+      {isActive === 1 ? (
+        <Dialog
+          buttons={
+            <>
+              <Button.DialogClose component={Button.BorderedGray}>
+                {commonText.cancel()}
+              </Button.DialogClose>
+            </>
+          }
+          header={treeText.addTree()}
+          onClose={() => setIsActive(0)}
+        >
+          <Ul className="flex flex-col gap-2">
+            {defaultTreeDefs.map((resource, index) => (
+              <li key={index}>
+                <Button.LikeLink onClick={(): void => handleClick(resource)}>
+                  {resource.name}
```

Comment:
regarding the typescript error you are getting.
the error is there to prevent accidentally displaying unlocalized content in the UI - everything that is shown in the UI should be localized, unless there is a good reason to be otherwise (i.e in schema config, we are displaying unlocalized field names)

once you refactored `./TreeDefaults.tsx` to localize them, you can explicitly tell TypeScript that this string is localized by doing this:

```suggestion
                  {localized(resource.name)}
```